docs/enhancements/metricsConfig_redesign.md

```diff
@@ -0,0 +1,169 @@
+
+#### Additional label support
+Hive can manage a lot of clusters, so we usually avoid labelling any metric with cluster name or namespace to keep the cardinality in check. So, in addition to the labels hive reports for a metric, some admins might want extra labels.
+For example, if admins can identify that a metric corresponds to a "managed" cluster, they can apply effective filters for their observability and can observe trends across only managed clusters, and also customize their alerts.
+
+- **Pros for this design**: The process of applying the labels to clusterdeployment, configuring those labels in hiveconfig to be reported with the metric, and using them as filters for alerts and grafana queries - is all in the hands of the admins/users and does not require any code change.
+It also puts the onus of cardinality on the admins.
+- **Cons we would like to overcome**: 
+  - The way the code is written right now - we have a hardcoded hack for cluster-type label as mandatory if no additional labels are configured, and, most importantly, any additional labels configured will be applied to all the metrics which have optional label support, even though they might not be needed for all.
+  We're essentially worsening the cardinality this way, and it makes adding this support to more metrics tricky. 
+  - There's also the caveat of parsing through fixed and optional labels - ensuring there's no overlap and maintaining the order for certain metrics which do not accept map of label-value while observing, are some complications that currently exist and probably will not be fixed as a part of this enhancement.
+
+
+### Goals
+- Make all Hive metrics optional. By this design, no metric is to be reported by default unless it is explicitly configured in HiveConfig.
+- Allow configuring additional labels, duration based threshold and expression matching support for all metrics that are related to cluster deployment, via HiveConfig.
+- Enforce naming convention for duration based metrics - all metrics that report duration and allow customization for duration-based threshold must be named `*_seconds`.
+- Move away from throwing a panic if metricsConfig fails validation, update HiveConfig.status.Conditions instead
+
+### Non-Goals
+- Non-cluster deployment related metrics - like metrics reported by HiveOperator, or metrics that do not have the related cluster deployment available while reporting (for ex, some syncset, selectorsyncset and clusterpool metrics) will not support additional label and expression matching. 
+- Adding labels only refers to the labels propagated from the Cluster Deployment, any other strings as labels cannot be configured this way.
```

Comment:
Not sure exactly how/where to put this, but I'd like to mention that, while the non-goals are _currently_ non-goals, it is a goal that, if we *did* decide to do those things in the future, it would be easy to expand this design to accommodate them. In fact, it may be worth calling out in the `Current implementations` section that a con of the old design that we would like to overcome is its non-expandability.